### ISSUE TITLE
PHP warnings on PHP 8.1

### ISSUE THREAD

**Issue by gino0631** [mijora/omniva-woocommerce#131]
The following warnings are logged when running on PHP 8.1:
```
PHP Warning:  Attempt to read property "children" on array in /var/www/html/wp-content/plugins/omniva-woocommerce/core/class-shipping-method.php on line 1142
PHP Warning:  Attempt to read property "term_id" on array in /var/www/html/wp-content/plugins/omniva-woocommerce/core/class-shipping-method.php on line 1143
PHP Warning:  Attempt to read property "name" on array in /var/www/html/wp-content/plugins/omniva-woocommerce/core/class-shipping-method.php on line 1143
```

Plugin version: 1.12.0
PHP version: 8.1.13

**Comment by markakk** [comment]
Hello,

We still can't adapt the module to PHP 8.1 yet, because at this moment the latest version of Wordpress (6.1.1) still has problems with this PHP version.

**Comment by gino0631** [comment]
@markakk are these WordPress problems related to the issues in your plugin?

**Comment by markakk** [comment]
@gino0631 Wordpress problems are preventing us from fully testing the Omniva plugin, so we decided to wait until Wordpress works better with PHP 8.1

**Comment by gino0631** [comment]
@markakk maybe it would be possible to look just into this particular issue, as it wasn't caused by PHP 8.1 itself - they only converted some notices into warnings, so you'll be able to reproduce the issue after setting `WP_DEBUG` to `true` even with PHP 7.4.

**Comment by markakk** [comment]
As we could find, we fixed the problems with PHP 8.1. Plugin version 1.12.3 should now work.
